lib/unit-ruby/util/connection.rb

```diff
@@ -1,5 +1,10 @@
 require 'faraday'
-require 'faraday_middleware'
+begin
+  require 'faraday/retry'
+rescue LoadError
+  puts "Add `gem 'faraday-retry'` to your Gemfile"
+  raise
+end
```

Comment:
From what I can tell, `faraday-retry` requires `faraday` version `~> 2.0`. Do you think we might need to do something to require either 'faraday/retry' or 'faraday-middleware' depending on the faraday version?